### ISSUE TITLE
[BUG-REPORT] export_partitioned seems memory hungry

### ISSUE THREAD

**Issue by spartako** [vaexio/vaex#1237]
**Description**

Running export_partitioned on a sufficiently large (long?) dataset makes my machine OOM:
```
import vaex
df = vaex.example()[['id', 'x']]
df_long = df.concat(*[df for _ in range(3000)])
df_long.export_parquet(path="example.parquet")
```
We can check the size of that 
```
> du -h example.parquet
5.0G	example.parquet
```
This will consume much more than 5GB of RAM:
```
dff = vaex.open("example.parquet")
dff.export_partitioned("test", by=['id'])
```

**Software information**
 - Vaex version (`import vaex; vaex.__version__)`: {'vaex-core': '4.0.0a19', 'vaex-hdf5': '0.7.0a6'}
 - Vaex was installed via: pip 
 - OS: ubuntu20 / docker

**Comment by maartenbreddels** [comment]
Hi,

thanks for the report. You are correct, this takes `len(df) * 3000 * len(df.id.unique()) //1000**3` bytes of memory, since each dataframe keeps a cache for the filter. However, instead of iterating over the dataframes, we instantiated them all. This is fixed in #1260

**Comment by spartako** [comment]
Awesome, thank you!
